table/src/main/java/org/enso/table/data/column/storage/DoubleStorage.java

```diff
@@ -98,4 +102,37 @@ public DoubleStorage mask(BitSet mask, int cardinality) {
     }
     return new DoubleStorage(newData, cardinality, newMissing);
   }
+
+  @Override
+  public Storage orderMask(int[] positions) {
+    long[] newData = new long[positions.length];
+    BitSet newMissing = new BitSet();
+    for (int i = 0; i < positions.length; i++) {
+      if (positions[i] == Index.NOT_FOUND || isMissing.get(positions[i])) {
+        newMissing.set(i);
+      } else {
+        newData[i] = data[positions[i]];
+      }
+    }
+    return new DoubleStorage(newData, positions.length, newMissing);
+  }
+
+  @Override
+  public Storage countMask(int[] counts, int total) {
+    long[] newData = new long[total];
+    BitSet newMissing = new BitSet();
+    int pos = 0;
+    for (int i = 0; i < counts.length; i++) {
+      if (isMissing.get(i)) {
+        for (int j = 0; j < counts[i]; j++) {
+          newMissing.set(pos++);
+        }
```

Comment:
Why not do it like for Booleans?
```suggestion
       newMissing.set(pos, pos + counts[i]);
       pos += counts[i];
```